### ISSUE TITLE
Segmentation fault when forwarding to Quassel

### ISSUE THREAD

**Issue by lazyb0nes** [janikrabe/oidentd#17]
HI there,
I recently noticed that oidentd is crashing.
Here is some information on what ive found in the syslog as well as versiuon information.

```text
oidentd 2.3.1 by Janik Rabe <oidentd@janikrabe.com>
Originally written by Ryan McCabe <ryan@numb.org>
https://oidentd.janikrabe.com

Build information:
        Kernel driver: linux.c
        Needs kmem access: No
        Needs root access: No
        Debugging build: No
        Masquerading support: Yes
        IPv6 support: Yes
        Linux libcap-ng support: Yes
        Linux libnfct support: Yes
        UDB library support: No
```

```text
Aug 16 21:12:03 <hostname> oidentd[19300]: Connection from weber.freenode.net (162.213.39.42):34303
Aug 16 21:12:03 <hostname> oidentd[19300]: Caught SIGSEGV; please report this to oidentd@janikrabe.com
Aug 16 21:12:43 <hostname> oidentd[19304]: Connection from weber.freenode.net (162.213.39.42):57553
Aug 16 21:12:43 hostname oidentd[19304]: [weber.freenode.net] Successful lookup: 35590 , 6665 : quasselcore (quasselcore)
Aug 16 21:13:02 host quasselcore: [Warn ] CoreNetwork::doAutoReconnect(): Cannot reconnect while not being disconnected!
Aug 16 21:15:01 <hostname> CRON[19325]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
```

I am running Ubuntu 1804 LTS

**Comment by janikrabe** [comment]
Thanks for the report!

Summarizing a discussion on IRC: @lazyb0nes and I are both unable to reproduce this issue. I'm leaving it open because it could indicate a serious problem. I don't see anything suspicious in the forwarding code, but have not performed a full audit.

If anyone else runs into this issue and is able to get a `--debug` log, that would be much appreciated.

Configuration file for future reference:

```text
/*
 * This is the system-wide configuration file for oidentd.
 * This file provides defaults for users. The settings in
 * this system-wide file can be overridden by the per-user
 * configuration files of users who have been granted
 * sufficient capabilities through this file.
 *
 * See oidentd.conf(5) for more information.
 */

default {
        default {
                # Capabilities are disabled by default
                deny spoof
                deny spoof_all
                deny spoof_privport
                deny random
                deny random_numeric
                deny numeric
                deny hide
                deny forward
        }
}

/*
 * It is recommended that you hide any connections owned by
 * the superuser. Uncomment the following lines to do this.
 */

#user root {
#       default {
#               # Uncomment only one of the following two lines.
#               force hide            # Either report an error ...
#               force reply "nobody"  # ... or spoof ident responses
#       }
#}
user quasselcore {
        default {
                force forward ::1 10113
        }
}
```

**Comment by janikrabe** [comment]
Still unable to reproduce and I haven't received any similar reports; closing for now.

**Comment by janikrabe** [comment]
This issue seems to have affected builds with libnetfilter_conntrack on kernels without a connection tracking file. A fix will be included in oidentd 2.3.2. Thanks for the report!
